pkgs/python-packages/joinmarket/default.nix

```diff
@@ -0,0 +1,119 @@
+{ version
+, src
+, format
+, lib
+
+# TODO: Is it necessary to use this deps check?
+#, buildPythonPackageWithDepsCheck
+, buildPythonPackage
+, pythonOlder
+, pythonAtLeast
+, pythonRelaxDepsHook
+, pytestCheckHook
+, setuptools
+, fetchurl
+, chromalog
+, cryptography
+, service-identity
+, twisted
+, txtorcon
+, python-bitcointx
+, argon2_cffi
+, autobahn
+, bencoderpyx
+, klein
+, mnemonic
+, pyjwt
+, werkzeug
+, libnacl
+, pyopenssl
+}:
+
+#buildPythonPackageWithDepsCheck rec {
+buildPythonPackage rec {
+  pname = "joinmarket";
+  inherit version src format;
+
+  # From v0.9.11, the Python older than v3.8 is not supported. Python v3.12 is
+  # still not supported.
+  disabled = (pythonOlder "3.8") || (pythonAtLeast "3.13");
+
+  nativeBuildInputs = [
+    setuptools
+    pythonRelaxDepsHook
+  ];
+
+  propagatedBuildInputs = [
+    # base jm packages
+    chromalog
+    cryptography
+    service-identity
+    twisted
+    txtorcon
+
+    # jmbitcoin
+    python-bitcointx
+
+    # jmclient
+    argon2_cffi
+    autobahn
+    bencoderpyx
+    klein
+    mnemonic
+    pyjwt
+    werkzeug
+
+    # jmdaemon
+    libnacl
+    pyopenssl
+
+  ];
+
+  # FIXME: Try to use twisted and service-identity from nixpkgs-23.11 (there are lower stable versions)?
+  # E.g. alternative approach
+  #postPatch = ''
+  #  substituteInPlace pyproject.toml \
+  #    --replace-warn 'twisted==23.10.0' 'twisted==24.3.0' \
+  #    --replace-warn 'service-identity==21.1.0' 'service-identity==24.1.0' \
+  #    --replace-warn 'cryptography==41.0.6' 'cryptography==42.0.5'
+  #'';
+  pythonRelaxDeps = [
```

Comment:
I am still not completely sure what is the best approach here. The `pythonRelaxDeps` seems to work fine.

Reply:
See [here ](https://github.com/erikarvstedt/nix-bitcoin/commit/c2133b6336d6748b3cc78149f47e3c5d18af21a8) for an explanation why `substituteInPlace` should be preferred over `pythonRelaxDeps`.